sql/parser/eval.go

```diff
@@ -535,6 +728,15 @@ var cmpOps = map[cmpArgs]func(Datum, Datum) (Datum, error){
 	cmpArgs{EQ, floatType, floatType}: func(left Datum, right Datum) (Datum, error) {
 		return DBool(left.(DFloat) == right.(DFloat)), nil
 	},
+	cmpArgs{EQ, dateType, dateType}: func(left Datum, right Datum) (Datum, error) {
+		return DBool(left.(DDate).Equal(right.(DDate).Time)), nil
+	},
+	cmpArgs{EQ, timestampType, timestampType}: func(left Datum, right Datum) (Datum, error) {
```

Comment:
postgres does use use SET TIME ZONE to allow you present the timestamps

mydb=# select * from tzone
mydb-# ;
           k            
------------------------
 2015-09-01 00:00:00-04
 2015-09-01 01:00:00-04
 2015-09-01 02:00:00-04
 2015-09-01 03:00:00-04
 2015-08-31 20:00:00-04
 2015-09-01 04:00:00-04
(6 rows)

mydb=# SET TIME ZONE -7
mydb-# ;
SET
mydb=# select * from tzone
;
           k            
------------------------
 2015-08-31 21:00:00-07
 2015-08-31 22:00:00-07
 2015-08-31 23:00:00-07
 2015-09-01 00:00:00-07
 2015-08-31 17:00:00-07
 2015-09-01 01:00:00-07
(6 rows)